src/util-plugin.c

```diff
@@ -0,0 +1,75 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+#include "util-plugin.h"
+#include "suricata-plugin.h"
+
+#ifdef HAVE_DLFCN_H
+
+#include <dlfcn.h>
+
+static void InitPlugin(char *path)
+{
+    void *lib = dlopen(path, RTLD_LAZY);
+    if (lib != NULL) {
+        SCLogNotice("Loading plugin %s", path);
+        void (*registerf)(SCPlugin *) = dlsym(lib, "sc_plugin_register");
+        if (registerf == NULL) {
+            SCLogNotice("Plugin missing registration function: %s", path);
+        } else {
+            SCPlugin plugin;
+            memset(&plugin, 0, sizeof(plugin));
+            (*registerf)(&plugin);
+            BUG_ON(plugin.name == NULL);
+            BUG_ON(plugin.author == NULL);
+            BUG_ON(plugin.license == NULL);
```

Comment:
Should we check if the plugin license is compatible?

Reply:
Maybe some sort of opt-in check. Or opt-out flag. We should not prevent an end-user from using a plugin with an evil license.

Reply:
Yes, this seems like a control we should provide for the user (opt-in/opt-out)